app/jobs/scheduled/reindex_search.rb

```diff
@@ -1,201 +1,162 @@
 # frozen_string_literal: true
 
 module Jobs
-  # if locale changes or search algorithm changes we may want to reindex stuff
   class ReindexSearch < ::Jobs::Scheduled
     every 2.hours
 
-    CLEANUP_GRACE_PERIOD = 1.day.ago
-
     def execute(args)
-      @verbose = true if args && Hash === args && args[:verbose]
+      @verbose = args[:verbose]
+      @cleanup_grace_period = 1.day.ago
 
-      rebuild_problem_topics
-      rebuild_problem_posts
-      rebuild_problem_categories
-      rebuild_problem_users
-      rebuild_problem_tags
-      clean_post_search_data
-      clean_topic_search_data
+      rebuild_categories
+      rebuild_tags
+      rebuild_topics
+      rebuild_posts
+      rebuild_users
 
-      @verbose = nil
+      clean_topics
+      clean_posts
     end
 
-    def rebuild_problem_categories(limit: 500)
+    def rebuild_categories(limit: 500, indexer: SearchIndexer)
       category_ids = load_problem_category_ids(limit)
 
-      if @verbose
-        puts "rebuilding #{category_ids.length} categories"
-      end
+      puts "rebuilding #{category_ids.size} categories" if @verbose
 
       category_ids.each do |id|
         category = Category.find_by(id: id)
-        SearchIndexer.index(category, force: true) if category
+        indexer.index(category, force: true) if category
       end
     end
 
-    def rebuild_problem_users(limit: 10000)
-      user_ids = load_problem_user_ids(limit)
+    def rebuild_tags(limit: 1_000, indexer: SearchIndexer)
```

Comment:
Looks like this changed from `10_000` to `1_000`. Was that deliberate?

Reply:
Kinda... I feel like `1000` tags at a time should be more than enough. Should also leave some "room"  (eg. time) for other models.